app/renderer/js/main.ts

```diff
@@ -231,14 +233,37 @@ class ServerManagerView {
 		}
 	}
 
+	onEnd(): void {
+		const newServers: any[] | ServerOrFunctionalTab[] = [];
+		const tabMap: any = {};
```

Comment:
You should create a interface here for this, it's just:
```suggestion
interface TabMap {
    [key: number]: number;
}
const tabMap: TabMap = {};
```

Reply:
Also we should probably add a detailed comment here what `tabMap` stores and how it works. I used know the logic back almost a year ago and now looking at this I have no idea what this is which is the perfect reason to add a comment here.